spec/requests/users_controller_spec.rb

```diff
@@ -5254,32 +5255,101 @@ def revoke(user, skip_remote: false)
         URL:#{Discourse.base_url}/t/-/#{bookmark2.topic_id}
         END:VEVENT
         END:VCALENDAR
-      ICS
-    end
+        ICS
+      end
 
-    it "does not show another user's bookmarks" do
-      sign_in(user1)
-      get "/u/#{bookmark3.user.username}/bookmarks.json"
-      expect(response.status).to eq(403)
-    end
+      it "does not show another user's bookmarks" do
+        sign_in(user1)
+        get "/u/#{bookmark3.user.username}/bookmarks.json"
+        expect(response.status).to eq(403)
+      end
 
-    it "shows a helpful message if no bookmarks are found" do
-      bookmark1.destroy
-      bookmark2.destroy
-      bookmark3.destroy
-      sign_in(user1)
-      get "/u/#{user1.username}/bookmarks.json"
-      expect(response.status).to eq(200)
-      expect(response.parsed_body['bookmarks']).to eq([])
+      it "shows a helpful message if no bookmarks are found" do
+        bookmark1.destroy
+        bookmark2.destroy
+        bookmark3.destroy
+        sign_in(user1)
+        get "/u/#{user1.username}/bookmarks.json"
+        expect(response.status).to eq(200)
+        expect(response.parsed_body['bookmarks']).to eq([])
+      end
+
+      it "shows a helpful message if no bookmarks are found for the search" do
+        sign_in(user1)
+        get "/u/#{user1.username}/bookmarks.json", params: {
+          q: 'badsearch'
+        }
+        expect(response.status).to eq(200)
+        expect(response.parsed_body['bookmarks']).to eq([])
+      end
     end
 
-    it "shows a helpful message if no bookmarks are found for the search" do
-      sign_in(user1)
-      get "/u/#{user1.username}/bookmarks.json", params: {
-        q: 'badsearch'
-      }
-      expect(response.status).to eq(200)
-      expect(response.parsed_body['bookmarks']).to eq([])
+    context "for polymorphic bookmarks" do
+      class UserTestBookmarkSerializer < UserBookmarkBaseSerializer
+        def title
+          fancy_title
+        end
+
+        def fancy_title
+          @fancy_title ||= user.username
```

Comment:
```suggestion
          user.username
```

I don't think the memoization is necessary in tests.